src/api/transaction/sign.js

```diff
@@ -48,12 +48,14 @@ function computeSignature(txJSON, keypair) {
 }
 
 function sign(txJSON: {Account: string; SigningPubKey: string,
-    TxnSignature: string}, secret: string):
+    TxnSignature: string}, secret: string, isRegular = false):
```

Comment:
I think we should avoid the extra complexity of this parameter. We can just delete the `validate.addressAndSecret` and maybe we can just remove the address parameter below?